lib/hyacinth/adapters/external_identifier_adapter/datacite.rb

```diff
@@ -4,109 +4,136 @@ module Hyacinth
+          datacite_data.build_properties_update(doi_state)
+          rest_api = Datacite::RestApi::V2::Api.new(DATACITE[:rest_api], DATACITE[:user], DATACITE[:password])
+          rest_api_response = rest_api.put_dois(doi, datacite_data.generate_json_payload)
+          unless rest_api_response.status.eql? 200
+            Rails.logger.error "Did not update a DOI! Response Code: #{rest_api_response.status}." \
+                               "Response Body: #{rest_api_response.body}." \
+                               "Request Body: #{rest_api.most_recent_request_body}"
+            return
+          end
+          rest_api_response.body
         end
 
-        # Following method will make a request to the EZID server to update the target URL associated
-        # with the EZID DOI entry on the server.
-        # (See _target in http://ezid.cdlib.org/doc/apidoc.html#internal-metadata)
-        # To delete the target URL stored in the DOI server, use an empty string as the argument.
-        # Uses the modify identifier API call, returns true if metadata update was successful
-        # if not, returns false
+        # Following only updates the target url. Assumes the other required DataCite properties
+        # are already set. If not sure, use update_doi(doi, hyacinth_metadata, target_url, doi_state = nil)
+        # and supply required metadata
+        # return nil if unsuccessful, else returns the response body
         def update_doi_target_url(doi, target_url)
-          return false if doi.nil?
-          return false if target_url.nil?
-          # ApiSession#modify_identifier returns true if the response from the EZID server indicated
-          # success, else it returns false
-          api_session.modify_identifier(doi, _target: target_url, _status: IDENTIFIER_STATUS[:public])
+          datacite_data = Datacite::RestApi::V2::Data.new
+          datacite_data.prefix = DATACITE[:prefix]
+          datacite_data.url = target_url
+          datacite_data.build_properties_update
+          rest_api = Datacite::RestApi::V2::Api.new(DATACITE[:rest_api], DATACITE[:user], DATACITE[:password])
+          rest_api_response = rest_api.put_dois(doi, datacite_data.generate_json_payload)
+          unless rest_api_response.status.eql? 200
+            Rails.logger.error "Did not mint a DOI! Response Code: #{rest_api_response.status}." \
+                               "Response Body: #{rest_api_response.body}." \
+                               "Request Body: #{rest_api.most_recent_request_body}"
+            return
+          end
+          rest_api_response.body
         end
 
-        def tombstone_impl(id)
-          return false if id.nil?
-          # ApiSession#modify_identifier returns true if the response from the EZID server indicated
-          # success, else it returns false
-          api_session.modify_identifier(id, _status: IDENTIFIER_STATUS[:unavailable])
+        def map_metadata(hyacinth_metadata, datacite_data, target_url)
```

Comment:
If you want to preserve the separation of Hyacinth data and the Data wrapper, this method could return a hash based on `hyacinth_metadata`. Alternately, this method could move onto the Data wrapper, though that would be less encapsulated.

Reply:
Method now returns a hash.